### ISSUE TITLE
useSyncExternalStore should have a correct/frozen getServerSnapshot

### ISSUE THREAD

**Issue by TkDodo** [TanStack/query#4690]
I think we should also do this in v5

@Ephem

**Comment by Ephem** [comment]
Yeah, I think we _could_ probably do it in a minor, it doesn't break anything, but it will probably be easier to include together with the other work being done on hydration. :smiley:
